test/issue-3288.js

```diff
@@ -0,0 +1,74 @@
+const { tspl } = require('@matteo.collina/tspl')
+const { test, after } = require('node:test')
+const { createServer } = require('node:http')
+const { once } = require('node:events')
+
+const { RetryHandler, Client } = require('..')
+
+test('Issue #3288 - request with body (asynciterable)', async t => {
+    t = tspl(t, { plan: 6 })
+    const server = createServer()
+    const dispatchOptions = {
+      method: 'POST',
+      path: '/',
+      headers: {
+        'content-type': 'application/json'
+      },
+      body: (function * () {
+        yield 'hello'
+        yield 'world'
+      })()
+    }
+  
+    server.on('request', (req, res) => {
+      res.writeHead(500, {
+        'content-type': 'application/json'
+      })
+  
+      res.end('{"message": "failed"}')
+    })
+  
+    server.listen(0, () => {
+      const client = new Client(`http://localhost:${server.address().port}`)
+      const handler = new RetryHandler(dispatchOptions, {
```

Comment:
as in this RetryHandler, I've called it with arguments that are needed for the implementation  in lib/handler/retry-handler.js